ext/mysql2/violite.h

```diff
@@ -0,0 +1,280 @@
+/* Copyright (c) 2000, 2015, Oracle and/or its affiliates. All rights
```

Comment:
let's talk through this one before I address the rest of the stuff.  Yes, you're definitely right that this isn't part of the public api; I'm not particularly happy with the solution, but I looked for quite a bit and couldn't find any function, public or not, that would advise you about whether you had to block.  

Having said that, the history of this structure is that it hasn't had a breaking change since (maybe) 2104 or (probably) 2012: https://github.com/mysql/mysql-server/commits/5.7/include/violite.h